### PR TITLE
Categorical passthrough encoder training failure fix

### PR DESCRIPTION
**Pull request by abidwael**
Fix for ECD training failure when we set
```
defaults:
  category:
    encoder:
      type: passthrough
```

### CODE REVIEW

File: ludwig/encoders/generic_encoders.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 logger = logging.getLogger(__name__)
 
 
-@register_encoder("passthrough", [CATEGORY, NUMBER, VECTOR])
+@register_encoder("passthrough", [NUMBER, VECTOR])
```
Comment (dantreiman) [inline ludwig/encoders/generic_encoders.py:29]: I think I'm missing some context; what is the use case for this?

Categorical variables aren't continuous so I don't know why we would want a passthrough encoder for them.

**Review by dantreiman — CHANGES_REQUESTED** [review]

File: ludwig/encoders/generic_encoders.py (hunk 1 above)
Comment (abidwael) [reply ludwig/encoders/generic_encoders.py:29]: I don't think there's a new use case for this. We already support passthrough encoder for categorical features, but specifying it as an encoder in the config leads to an error.

Comment (justinxzhao) [reply ludwig/encoders/generic_encoders.py:29]: The passthrough encoder for category features is used by the gbm model type. CC: @jppgks

**Review by justinxzhao — APPROVED** [review]

File: ludwig/encoders/generic_encoders.py (hunk 1 above)
Comment (dantreiman) [reply ludwig/encoders/generic_encoders.py:29]: Aha, that explains it.  Makes sense for decision trees.

**Review by dantreiman — APPROVED** [review]
LGTM

**Pull request merged**